task/sealights-nodejs-oci-ta/0.1/sealights-nodejs-oci-ta.yaml

```diff
@@ -0,0 +1,161 @@
+      default: ""
+    - name: target-branch
+      type: string
+      description: "The name of the target branch for the pull request, typically the branch into which the changes will be merged (e.g., 'main', 'develop')."
+      default: "main"
+    - name: workspace-path
+      type: string
+      description: "Path to the directory that should be scanned"
+      default: "."
+  volumes:
+    - name: sealights-credentials
+      secret:
+        secretName: sealights-credentials
+    - name: workdir
+      emptyDir: {}
+  stepTemplate:
+    volumeMounts:
+      - mountPath: /opt/app-root
+        name: workdir
+      - name: sealights-credentials
+        mountPath: /usr/local/sealights-credentials
+  steps:
+    - name: use-trusted-artifact
+      image: quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:52f1391e6f1c472fd10bb838f64fae2ed3320c636f536014978a5ddbdfc6b3af
+      args:
+        - use
+        - $(params.SOURCE_ARTIFACT)=/opt/app-root/src
+    - name: sealights-nodejs-instrumentation
+      image: registry.access.redhat.com/ubi8/nodejs-$(params.nodejs-version):latest
+      workingDir: /opt/app-root/src
+      securityContext:
+        runAsUser: 0
+      env:
+        - name: COMPONENT
+          value: $(params.component)
+        - name: IS_FRONTEND
+          value: $(params.is-frontend)
+        - name: SCM
+          value: $(params.scm)
+        - name: SCM_PROVIDER
+          value: $(params.scm-provider)
+        - name: REPOSITORY_URL
+          value: $(params.repository-url)
+        - name: BRANCH
+          value: $(params.branch)
+        - name: REVISION
+          value: $(params.revision)
+        - name: PULL_REQUEST_NUMBER
+          value: $(params.pull-request-number)
+        - name: TARGET_BRANCH
+          value: $(params.target-branch)
+        - name: TEST_EVENT
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.labels['pipelinesascode.tekton.dev/event-type']
+        - name: WORKSPACE_PATH
+          value: $(params.workspace-path)
+      args: ["$(params.exclude[*])"]
+      script: |
+        #!/bin/bash
+        set -euo pipefail
+
+        export SEALIGHTS_TOKEN BUILD_NAME PATHS_EXCLUDED_ENUM
+        SEALIGHTS_TOKEN="$(cat /usr/local/sealights-credentials/token)"
+        BUILD_NAME="${REVISION}_$(date +'%y%m%d.%H%M')"
+        PATHS_EXCLUDED_ENUM="$(IFS=,; printf "%s," "$@" | sed 's/,$//')"
+
+        # Download jq in order to help determine the Agent version below
+        curl -Lso jq https://github.com/stedolan/jq/releases/download/jq-1.7.1/jq-linux64 && chmod +x jq && mkdir -p /opt/app-root/bin && mv jq /opt/app-root/bin/jq
+
+        # Installs Sealights Node.js agent.
+        # Fetch Agent version defined in Sealights dashboard settings
+        SL_VERSION=$(curl -X GET "https://redhat.sealights.co/api/v2/agents/slnodejs/recommended" -H "accept: application/json" -H "Authorization: Bearer $SEALIGHTS_TOKEN" -L | jq -r .agent.version)
```

Comment:
Could you file a follow-up story to build a sealights container and install the necessary dependencies at build time?

Reply:
👍  [done](https://issues.redhat.com/browse/KFLUXDP-158)